core/services/runs.go

```diff
@@ -86,6 +86,22 @@ func NewRun(
 	run.CreationHeight = models.NewBig(currentHeight)
 	run.ObservedHeight = models.NewBig(currentHeight)
 
+	if !job.MinPayment.IsZero() && job.MinPayment != nil && input.Amount != nil && job.MinPayment.Cmp(input.Amount) == -1 {
```

Comment:
I'd love to see this logic put into a little helper function to explain what it's doing e.g: `meetsMinimumMayment -> bool`.